src/compiler/scala/tools/nsc/settings/AbsSettings.scala

```diff
@@ -22,19 +22,19 @@ trait AbsSettings extends scala.reflect.internal.settings.AbsSettings {
   type Setting <: AbsSetting      // Fix to the concrete Setting type
   type ResultOfTryToSet           // List[String] in mutable, (Settings, List[String]) in immutable
   def errorFn: String => Unit
-  protected def allSettings: scala.collection.Set[Setting]
+  protected def allSettings: scala.collection.Map[String, Setting]
```

Comment:
It would be rude to say that settings are an enum. My Java friend loves enums and enum maps and sets. I don't know why. One particular enum of a domain object has grown, and he couldn't believe there were so many of them. I'm not sure why that matters. When do you say, "That's a lot of enums!"